src/org/traccar/database/PermissionsManager.java

```diff
@@ -300,7 +300,13 @@ public void checkGroup(long userId, long groupId) throws SecurityException {
     }
 
     public void checkDevice(long userId, long deviceId) throws SecurityException {
-        if (!Context.getDeviceManager().getUserItems(userId).contains(deviceId) && !getUserAdmin(userId)) {
+        if (!Context.getDeviceManager().getUserItems(userId).contains(deviceId)) {
+            checkDeviceManagerOrAdmin(userId, deviceId);
+        }
+    }
+
+    public void checkDeviceManagerOrAdmin(long userId, long deviceId) throws SecurityException {
```

Comment:
I don't think we need a separate method. Please implement it the same way it's implement in other places.

Reply:
Then We have to implement whole logic in the Device Resource's updateTotalDistance method or change the checkDeviceManagerOrAdmin method name?

Reply:
You don't need another method and you don't need to move the logic anywhere. Just look at the `BaseObjectResource` example.